### PR TITLE
[mono][infra] Fix mono-aot-cross build for linux-arm64 using biarch image

### PR DESCRIPTION
**Pull request by kotlarmilos**
## Description

This PR resolves the issue with the mono-aot-cross build on linux-arm64. The biarch image `cbl-mariner-2.0-cross-biarch-amd64-arm64` provides rootfs for both x64 and arm64 architectures. The runtime is built using the `ROOTFS_DIR` which points to `/crossrootfs/arm64` The `mono-aot-cross` is built using the `ROOTFS_HOST_DIR` which points to `/crossrootfs/x64`.

**Note:** This PR is one step closer to functional CI jobs, but doesn't fully fix them.

Contributes to https://github.com/dotnet/runtime/issues/90427

### CODE REVIEW

File: eng/pipelines/extra-platforms/runtime-extra-platforms-other.yml [hunk 1]
```diff
@@ -362,7 +362,9 @@ jobs:
     runtimeFlavor: mono
     platforms:
       - linux_x64
```
[the remark this answers is not in the document]
Comment (kotlarmilos) [reply eng/pipelines/extra-platforms/runtime-extra-platforms-other.yml:364]: Sorry for the late reply. Currently, the linux jobs are broken (more details in https://github.com/dotnet/runtime/issues/86324#issuecomment-1549896743), and we are working on a fix, but in the meantime we tested OSX (which should work). However, OSX machines are too slow, taking 7h to finish.

**Comment by kotlarmilos** [comment]
Official build job: https://dev.azure.com/dnceng/internal/_build/results?buildId=2261982&view=results

**Review by directhex — APPROVED** [review]

**Pull request merged**